### ISSUE TITLE
Crash on Setting PIN #2

### ISSUE THREAD

**Issue by HashEngineering** [HashEngineering/dash-wallet#106]
This has happened twice, both on Android 6.
One crash on 5.17.5
One crash on 5.18

This is not a new crash.

```
Today, 6:32 AM on app version 51800
Motorola Moto G (2nd Gen) (titan_umts), 1024MB RAM, Android 6.0
Report 1 of 2

java.lang.IllegalStateException: 
  at android.app.FragmentManagerImpl.checkStateLoss (FragmentManager.java:1413)
  at android.app.FragmentManagerImpl.enqueueAction (FragmentManager.java:1431)
  at android.app.BackStackRecord.commitInternal (BackStackRecord.java:687)
  at android.app.BackStackRecord.commit (BackStackRecord.java:663)
  at android.app.DialogFragment.dismissInternal (DialogFragment.java:301)
  at android.app.DialogFragment.dismiss (DialogFragment.java:267)
  at de.schildbach.wallet.ui.EncryptKeysDialogFragment$3$1$1.run (EncryptKeysDialogFragment.java:328)
  at android.os.Handler.handleCallback (Handler.java:746)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:148)
  at android.app.ActivityThread.main (ActivityThread.java:5443)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:728)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:618)
```

**Comment by HashEngineering** [comment]
Similar error, but different exception.

```
Today, 5:22 PM on app version 51800
ZTE Infinity X^2 mini (msm8226), 1024MB RAM, Android 4.4
Report 1 of 1
java.lang.NullPointerException: 
  at android.app.DialogFragment.dismissInternal (DialogFragment.java:287)
  at android.app.DialogFragment.dismiss (DialogFragment.java:258)
  at de.schildbach.wallet.ui.EncryptKeysDialogFragment$3$1$1.run (EncryptKeysDialogFragment.java:328)
  at android.os.Handler.handleCallback (Handler.java:733)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:136)
  at android.app.ActivityThread.main (ActivityThread.java:5050)
  at java.lang.reflect.Method.invokeNative (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:515)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:786)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:602)
  at dalvik.system.NativeStart.main (Native Method)
```

**Comment by HashEngineering** [comment]
Fixed with bb9d93e85a09141a61748a01848a7b75ee280d00.
